mindmaps-engine/src/main/java/io/mindmaps/engine/controller/VisualiserController.java

```diff
@@ -105,11 +109,11 @@ private String matchQuery(Request req, Response res) {
             final JSONArray halArray = new JSONArray();
 
             withGraph(graph).parseMatch(req.queryParams(REST.Request.QUERY_FIELD))
-                    .stream()
+                    .parallelStream()
```

Comment:
So it shouldnt be a problem.

But anyway I have made a new pull request with only strema() and when we will have time to profile everything again I will check if we can parallelise